File: openni/python/ni-hello.py
from openni import openni2

openni2.initialize()
uris = openni2.Device.enumerate_uris()

dev = openni2.Device.open_file(uris[0])
deviceInfo = dev.get_device_info()
print(f'Name: {deviceInfo.name}')
print(f'Uri: {deviceInfo.uri}')
print(f'USB Product ID: {deviceInfo.usbProductId}')
print(f'USB Vendor ID: {deviceInfo.usbVendorId}')
print(f'Vendor: {deviceInfo.vendor}')

dev.close()
openni2.unload()

File: openni/c++/ni-hello/ni-hello.cpp
#include <openni2/OpenNI.h>

int main()
{
	if (openni::OpenNI::initialize() != openni::STATUS_OK)
	{
		printf("Initialize Failed:%s\n", openni::OpenNI::getExtendedError());
		return -1;
	}

	openni::Array<openni::DeviceInfo> deviceList;
	openni::OpenNI::enumerateDevices(&deviceList);

	for (int i = 0; i < deviceList.getSize(); i++)
	{
		openni::DeviceInfo info = deviceList[i];
		printf("Name:%s\n", info.getName());
		printf("Uri:%s\n", info.getUri());
		printf("USB Product ID:%d\n", info.getUsbProductId());
		printf("USB Vendor ID:%d\n", info.getUsbVendorId());
		printf("Vendor:%s\n", info.getVendor());
	}

	openni::Device device;
	if (device.open(deviceList[0].getUri()) != openni::STATUS_OK)
	{
		printf("Cannot open device: %s\n", openni::OpenNI::getExtendedError());
		return -1;
	}

	device.close();
	openni::OpenNI::shutdown();
}
